[Gemma-4-26B]: torch.or_ fails with AttributeError: 'Tensor' object has no attribute 'or_'

### What happened?

When running op-level testing for torch_or_ operations (e.g., torch_or___1 and torch_or___191) with boolean inputs during model op-level testing for gemma-4-12B-it, the test fails during the CPU reference output computation stage. The op helper function _tensor_or_ attempts to call x.or_(other), which raises an AttributeError because torch.Tensor does not have an or_ attribute/method.

```
AttributeError: 'Tensor' object has no attribute 'or_'
```

### What did you expect to happen?

The _tensor_or_ op helper in op_registry.py should execute logical or bitwise OR on the input tensors using valid PyTorch operations (such as torch.bitwise_or(x, other) or x | other), allowing the reference output to be computed properly and enabling test validation to proceed on both CPU and Spyre devices.


### How can we reproduce it?

Step 1: Ensure config_yamls/gemma-4-26B-A4B-it_spyre.yaml is present. If not, regenerate it:
```
cd utils/model-opstest-yaml-generator
export HF_TOKEN=<your_token>
TEST_SCRIPT_FOLDER=./config_yamls python -m oplist.gemma4_26b.test_config_run_huggingface
```
Step 2: Run the specific failing test from the torch-spyre repo:
```
PYTORCH_TEST_CONFIG=tests/configs/model_ops_tests/gemma-4-26B-A4B-it_spyre.yaml \
TORCH_DEVICE_ROOT=$(pwd) \
PYTHONPATH=tests/models:tests \
pytest tests/models/test_model_ops_v2.py \
    --model model__gemma-4-26B-A4B-it_spyre \
    -k "torch_or___1" \
    -v
```
Or run the exact test case directly:
```
python tests/models/test_model_ops_v2.py \
    TestSpyreModelOpsCPU.test_model_ops_db_torch_or___1_cpu_bool
```

**Minimal reproducer (standalone):**
```
import torch
# Failing reference implementation inside op_registry.py (_tensor_or_)
x = torch.zeros((1, 34), dtype=torch.bool)
other = torch.zeros((1, 34), dtype=torch.bool)
# AttributeError: 'Tensor' object has no attribute 'or_'
result = x.or_(other)
```

### Any environmental details we need to know?

Ensure the tests are executed within the torch-spyre environment.

### Anything else we need to know?

_No response_

### Relevant log output

```Shell
[TAGS = tensorsoncpu torch.or_.1_spyre]
[INPUT SHAPES]
  arg[0]: Tensor(shape=[1, 34], dtype=torch.bool, stride=[34, 1])
  arg[1]: Tensor(shape=[1, 34], dtype=torch.bool, stride=[34, 1])
_________ TestSpyreModelOpsCPU.test_model_ops_db_torch_or___1_cpu_bool _________

self = <models.test_model_ops_v2.TestSpyreModelOpsCPU testMethod=test_model_ops_db_torch_or___1_cpu_bool>
device = 'cpu', dtype = torch.bool
op = ModelOpInfo(name='torch_or___1', ref=None, aliases=(), variant_test_name='', op=<function _tensor_or_ at 0x7f436eea0ae...ght=False, is_factory_function=False, skip_correctness_check_compile_vs_eager=False, has_nondeterministic_output=False)

    @ops(model_ops_db)
    def test_model_ops_db(self, device: str, dtype: torch.dtype, op: ModelOpInfo):
...
        # Run
        fn = adapter.fn
        try:
            with torch.no_grad():
>               ref_out = fn(cpu_sample.input, *cpu_sample.args, **cpu_sample.kwargs)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

tests/models/test_model_ops_v2.py:498: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

x = tensor([[False, False, False, ...]])
other = tensor([[False, False, False, ...]])

    def _tensor_or_(x: torch.Tensor, other):
>       return x.or_(other)
               ^^^^^
E       AttributeError: 'Tensor' object has no attribute 'or_'
E       
E       To execute this test, run the following from the base repo dir:
E           python test_model_ops_v2.py TestSpyreModelOpsCPU.test_model_ops_db_torch_or___1_cpu_bool

tests/models/op_registry.py:288: AttributeError
```

--- a/tests/models/op_registry.py
+++ b/tests/models/op_registry.py
@@ -318,7 +318,10 @@ def _tensor_and_(x: torch.Tensor, other):
 
 
 def _tensor_or_(x: torch.Tensor, other):
-    return x.or_(other)
+    # torch.Tensor has no ``or_`` method; the in-place spelling of bitwise or
+    # is ``bitwise_or_`` (identical to logical or on the bool inputs these
+    # tests use). Kept in-place to match the adapter's is_inplace=True.
+    return x.bitwise_or_(other)
 
 
 def _tensor_copy_(x: torch.Tensor, source: torch.Tensor):

--- a/tests/oot_framework/oot_test_config_models.py
+++ b/tests/oot_framework/oot_test_config_models.py
@@ -953,6 +953,19 @@ class InputsEdits(BaseModel):
                         f"config so the constructor arg uses 'config_path' + "
                         f"'config_kwargs' instead of a bare '<config:...>' value."
                     )
+                # A dtype recorded as its repr (e.g. 'torch.bfloat16'). Left as
+                # a string it reaches the op as a positional arg and is
+                # misinterpreted -- x.to('torch.bfloat16') parses it as a
+                # DEVICE string and raises. kwargs already resolve dtypes;
+                # positional values get the same treatment. Gated on a known
+                # dtype name so device strings ('cpu', 'cuda:0') and any other
+                # 'torch.'-prefixed value fall through unchanged.
+                if (
+                    isinstance(val, str)
+                    and val.startswith("torch.")
+                    and val.removeprefix("torch.") in _VALID_DTYPE_STRINGS
+                ):
+                    val = _resolve_dtype_str(val)
                 if (
                     test_device is not None
                     and op_name == "torch.to"

